dm/pkg/checker/privilege.go

```diff
@@ -112,8 +113,16 @@ func (pc *SourceReplicatePrivilegeChecker) Check(ctx context.Context) *Result {
 		markCheckError(result, err)
 		return result
 	}
-
-	verifyPrivileges(result, grants, replicationPrivileges)
+	replicationPrivileges := map[mysql.PrivilegeType]struct{}{
+		mysql.ReplicationClientPriv: {},
+		mysql.ReplicationSlavePriv:  {},
+	}
```

Comment:
It seems that an account with `super` privilege but without `replication client` or `replication slave` should also pass out pre-check.
```sql
mysql> show grants;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Grants for dm_full@%                                                                                                                                                       |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, REFERENCES, ALTER, SHOW DATABASES, SUPER, CREATE VIEW, SHOW VIEW, CREATE USER, TRIGGER ON *.* TO 'dm_full'@'%' |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> show master status;
+------------------+----------+--------------+------------------+---------------------------------------------+
| File             | Position | Binlog_Do_DB | Binlog_Ignore_DB | Executed_Gtid_Set                           |
+------------------+----------+--------------+------------------+---------------------------------------------+
| mysql-bin.000013 |     7877 |              |                  | 3eed27e3-3237-11ec-876d-0242ac110002:1-4440 |
+------------------+----------+--------------+------------------+---------------------------------------------+
1 row in set (0.00 sec)

mysql> show slave status;
Empty set (0.00 sec)
```

Reply:
ping @lichunzhu   now i thinks this pr is ready to merge, do you have more suggestions ?

Reply:
> ping @lichunzhu now i thinks this pr is ready to merge, do you have more suggestions ?

ping @lichunzhu